rcswitch/__init__.py

```diff
@@ -17,118 +23,126 @@
-				# check  if rc switch is installed at the specified path on remote host
-				self.fileStat = subprocess.check_output(shlex.split('sshpass -p {} ssh {}@{} stat -c %a {}'.format(rcswitch_password, rcswitch_user, rcswitch_host, self.rcswitch_dir)), stderr=DEVNULL).decode('utf8')
-				self.rcswitch_dir = ('sshpass -p {} ssh {}@{} {}'.format(rcswitch_password, rcswitch_user, rcswitch_host, self.rcswitch_dir))
-				self.logger.info('RCswitch: Using {} as host.'.format(rcswitch_host))
-			except subprocess.CalledProcessError as e:
-				self.setupOK = False
-				# give user hint where the problem is located
-				try:
-					if (user == rcswitch_user):
-						self.logger.error('RCswitch: send file of RCswitch not found at {} on {}. Check if RCswitch is installed correctly at specifed path on {}. System returned: {}'.format(self.rcswitch_dir, rcswitch_host, rcswitch_host, e))
-					else:
-						self.logger.error('RCswitch: send file of RCswitch not found at {} on {}. Additional problem with user authentication possible. System returned: {}'.format(self.rcswitch_dir, rcswitch_host, e))
-				except UnboundLocalError as e:
-					self.logger.error('RCswitch: Cannot connect to {}. Check rcswitch_host, rcswitch_user and rcswitch_password are set (correctly). Ensure SSH server is running on {}. System returned: {}'.format(rcswitch_host, rcswitch_host, e))
-		else:
-			# check if rc switch is installed at the specified path
-			if not os.path.isfile('{}/send'.format(self.rcswitch_dir)):
-				self.logger.error('RCswitch: send file of RCswitch not found at {} on localhost. Check path, if RCswitch is installed correctly on target, and correct format of v4 IP adress (in case rcswitch_host is defined).'.format(self.rcswitch_dir))
-				self.setupOK = False
-			else:
-				self.logger.info('RCswitch: setup on localhost OK')
+		Plugins have to use the new way of getting parameter values:
+		use the SmartPlugin method get_parameter_value(parameter_name). Anywhere within the Plugin you can get
+		the configured (and checked) value for a parameter by calling self.get_parameter_value(parameter_name). It
+		returns the value in the datatype that is defined in the metadata.
+		"""
+		# Call init code of parent class (SmartPlugin)
+		super().__init__()
+		# Initialization code goes here
+		# internal member variables
+		self._gpio = int(self.get_parameter_value('rcswitch_gpio'))
+		self._send_duration = float(self.get_parameter_value('rcswitch_sendDuration'))
+		# create threading.Lock() obj 
+		self._lock = threading.Lock()
 		
-		# setup semaphore
-		self.lock = threading.Lock()
+		# On initialization error use:
+		#   self._init_complete = False
+		#   return
+		
+		self.webif_pagelength = self.get_parameter_value('webif_pagelength')
```

Comment:
Das ist ein Arbeitsstand von OnkelAndy. Das ist hier nicht (mehr) notwendig. Das Handling von webif_pagelength erfolgt vollständig in ```webif/__init__.py``` (Siehe auch dev/sample_plugin)

Reply:
Ich hatte mich eigentlich an dem dev/ sample_plugin orientiert.
Ist die Vorlage auf dem aktuellen Stand?

Reply:
erledigt in PR für `rcs1000n`